network/api.go

```diff
@@ -242,6 +305,7 @@ func (h *Handler) initMux() {
 		if err != nil {
 			msg := err.Error()
 			sdk.EncodeResponse(w, NewErrorResponse(msg), msg)
+			return
```

Comment:
This is a bug, but isn't directly related to your changes.
It might be helpful to note that it was changed in the commit message